src/fields/fq_target.rs

```diff
@@ -107,27 +107,64 @@ impl<F: RichField + Extendable<D>, const D: usize> FqTarget<F, D> {
     pub fn is_equal(&self, builder: &mut CircuitBuilder<F, D>, rhs: &Self) -> BoolTarget {
         let a_limbs = self.target.value.limbs.iter().map(|x| x.0).collect_vec();
         let b_limbs = rhs.target.value.limbs.iter().map(|x| x.0).collect_vec();
-        assert_eq!(a_limbs.len(), b_limbs.len());
 
-        let terms = a_limbs
+        if a_limbs.len() == b_limbs.len() {
+            self.compare_same_length(builder, &a_limbs, &b_limbs)
+        } else {
+            self.compare_different_length(builder, &a_limbs, &b_limbs)
+        }
+    }
+
+    fn compare_same_length(
```

Comment:
Sorry I fail to see why the two functions are different - if you give same length inputs to the `compare_different_lengths` then it should be returning the same no ? 
I think the only change required is to move `if lhs.len() < rhs.len()` to `if lhs.len() <= rhs.len()`.
In case it's equal `  for &limb in longer.iter().skip(shorter.len()) {` will never run so it'll give the same results no ?

TLDR do we really need this new function ?

Reply:
Makes sense, it could be simplified as you said. Fixed in commit 698cc66, thanks!